java/src/main/java/com/cybersource/ws/client/SecurityUtil.java

```diff
@@ -190,50 +194,59 @@ private static void readAndStoreCertificateAndPrivateKey(MerchantConfig merchant
      * @throws SignException
      */
     public static Document handleMessageCreation(Document signedDoc, String merchantId, Logger logger) throws SignEncryptException, SignException{
-        
+
         logger.log(Logger.LT_INFO, "Encrypting Signed doc ...");
-        
-        WSSecHeader secHeader = new WSSecHeader();
+
+        WSSecHeader secHeader = new WSSecHeader(signedDoc);
         try {
-            secHeader.insertSecurityHeader(signedDoc);
+            secHeader.insertSecurityHeader();
         } catch (WSSecurityException e) {
             logger.log(Logger.LT_EXCEPTION, "Exception while adding document in soap securiy header for MLE");
             throw new SignException(e);
         }
-        
-        WSSecEncrypt encrBuilder = new WSSecEncrypt();
+
+        //Must initialize xml-security library correctly before use it
+        Init.init();
+
+        WSSecEncrypt encrBuilder = new WSSecEncrypt(secHeader);
         //Set the user name to get the encryption certificate.
         //The public key of this certificate is used, thus no password necessary. The user name is a keystore alias usually.
         encrBuilder.setUserInfo(identities.get(SERVER_ALIAS).getKeyAlias());
-        
+
         /*This is to reference a public key or certificate when signing or encrypting a SOAP message.
          *The following valid values for these configuration items are:
          *IssuerSerial (default),DirectReference[BST],X509KeyIdentifier,Thumbprint,SKIKeyIdentifier,KeyValue (signature only),EncryptedKeySHA1 (encryption only)
          */
         encrBuilder.setKeyIdentifierType(WSConstants.X509_KEY_IDENTIFIER);
-        
+
         //This encryption algorithm is used to encrypt the data.
         encrBuilder.setSymmetricEncAlgorithm(WSConstants.AES_256);
-        
+
         //Sets the algorithm to encode the symmetric key. Default is the WSConstants.KEYTRANSPORT_RSAOEP algorithm.
         //encrBuilder.setKeyEnc(WSConstants.KEYTRANSPORT_RSAOEP);
-        
-        
+
+
         //Create signed document
         //Document signedDoc = createSignedDoc(workingDocument,senderAlias,password,secHeader);
-        
+
         Document signedEncryptedDoc;
         try {
             //Builds the SOAP envelope with encrypted Body and adds encrypted key.
             // If no external key (symmetricalKey) was set ,generate an encryption
             // key (session key) for this Encrypt element. This key will be
             // encrypted using the public key of the receiver
-            signedEncryptedDoc = encrBuilder.build(signedDoc, localKeyStoreHandler, secHeader);
+            SecretKey symmetricKey;
+            try {
```

Comment:
Since there is no external key has set, I've generated a new key using "AES" Algorithm.